components/pagination/spec/Pagination.ts

```diff
@@ -0,0 +1,11 @@
+import { createElement as h } from 'react';
+import { mount } from '@not-govuk/component-test-helpers';
+import Pagination from '../src/Pagination';
+
+describe('Pagination', () => {
+  describe('when given valid props', () => {
+    const component = mount(h(Pagination, { heading: 'My heading' }, 'Child'));
+
+    it('renders', () => undefined);
```

Comment:
This isn't a simple component, I think there _really_ needs to be some proper tests here:
- counting the displayed items
- checking the page is correct
- checking the controls/links are present (or not at the start/end)
- does SSR work (Server-Side Rendering) _AND_ isomorphism
- can sub-pages be accessed by URL
- what about 0 items/empty table

You likely won't be able to achieve all of these checks here at the unit level, you will probably have to setup some interaction testing (e.g. using [Cypress](https://www.cypress.io/))

Reply:
> does SSR work (Server-Side Rendering) AND isomorphism
I think that's out of scope for the unit tests.

> can sub-pages be accessed by URL
Is that in scope for this component? Or is this just a visual display?
I think there are a lot of questions around what this component should actually be.

>You likely won't be able to achieve all of these checks here at the unit level, you will probably have to setup some interaction testing (e.g. using Cypress)
Cypress would be an application testing thing. If we want to test interaction here (do we have any?) then we should use enzyme for that.

Reply:
I've added a lot more tests for this component. Not sure if it's as comprehensive as you are expecting but hopefully getting there @RobinKnipe?